### PR TITLE
Improved battle debug number value changing

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
Originally utilized as part of #1965, as that value changer is based off BE's.
Instead of just moving up and down the number values, it increases the overall value by the selected digit. Eg: If the number 099 has the 3rd digit selected and the user presses Up, the value will increase to 100 instead of changing to 090.

## **Discord contact info**
AsparagusEduardo#6051

### CODE REVIEW

File: src/battle_debug.c [hunk 1]
```diff
@@ -1811,14 +1811,40 @@ static bool32 TryMoveDigit(struct BattleDebugModifyArrows *modArrows, bool32 mov
...
     {
         if (charDigits[modArrows->currentDigit] == CHAR_9)
+        {
             charDigits[modArrows->currentDigit] = CHAR_0;
+            for (i = modArrows->currentDigit - 1; i >= 0; i--)
+            {
+                if (charDigits[i] == CHAR_9)
+                    charDigits[i] = CHAR_0;
+                else
+                {
```
Comment (ultima-soul) [inline src/battle_debug.c:1821]:
```suggestion
                if (charDigits[i] == CHAR_9)
                {
                    charDigits[i] = CHAR_0;
                }
                else
                {
```

File: src/battle_debug.c [hunk 2]
```diff
@@ -1811,14 +1811,40 @@ static bool32 TryMoveDigit(struct BattleDebugModifyArrows *modArrows, bool32 mov
...
     {
         if (charDigits[modArrows->currentDigit] == CHAR_0)
+        {
             charDigits[modArrows->currentDigit] = CHAR_9;
+            
+            for (i = modArrows->currentDigit - 1; i >= 0; i--)
+            {
+                if (charDigits[i] == CHAR_0)
+                    charDigits[i] = CHAR_9;
+                else
```
Comment (ultima-soul) [inline src/battle_debug.c:1840]:
```suggestion
            for (i = modArrows->currentDigit - 1; i >= 0; i--)
            {
                if (charDigits[i] == CHAR_0)
                {
                    charDigits[i] = CHAR_9;
                }
                else
```

File: src/battle_debug.c [hunk 3]
```diff
@@ -1811,14 +1811,40 @@ static bool32 TryMoveDigit(struct BattleDebugModifyArrows *modArrows, bool32 mov
...
+                {
+                    charDigits[i]--;
+                    break;
+                }
+            }
+        }
+
         else
```
Comment (ultima-soul) [inline src/battle_debug.c:1848]:
```suggestion
        else
```

**Review by ultima-soul — CHANGES_REQUESTED** [review]
Just style consistency issues.

**Comment by AsparagusEduardo** [comment]
Ready for re-review

**Review by ultima-soul — APPROVED** [review]

**Pull request merged**